src/conda_package_handling/validate.py

```diff
@@ -29,3 +35,74 @@ def validate_converted_files_match(src_file_or_folder, subject, reference_ext=""
                     elif os.stat(absfile).st_size != os.stat(destpath).st_size:
                         mismatch_size.add(rp)
     return src_file_or_folder, missing_files, mismatch_size
+
+
+def hash_fn():
+    return hashlib.blake2b()
+
+
+IGNORE_FIELDS = {
+    "uid",
+    "gid",
+    "mtime",
+    "uname",
+    "gname",
+    "chksum",
+}  #: ignore if not strict
+
+
+def validate_converted_files_match_streaming(
+    src: str | Path, reference: str | Path, *, strict=True
```

Comment:
We cannot use this yet as we still need to support `>=3.7` (https://docs.python.org/3/library/stdtypes.html#types-union)

Reply:
I think you've noticed we can, with from future import annotations